### DISCUSSION TITLE
Constructed mesh is black

### DISCUSSION THREAD

**Discussion by DevAB-Git** [isl-org/Open3D#4973] [General]
Hi,
I'm trying to create a mesh for a simple environment (i.e. playpen, ROS noetic and Gazebo). I used 10 pcd files (recorded using HDL-32E lidar) to create the mesh environment by using the following steps:

1- Remove radius outliers (nb_points=10, radius=0.8) from pcd files and save as ply files

2- Register ply files using point-to-plane ICP and pose graph optimization

3- Combine the ply files. Apparently, the combined cloud looks good (see combined_plys.png).

4- Reconstruct the mesh environment using poisson reconstruction (depth=14). The resultant mesh file shows a black rectangle only (see front.png). The flipped side show a kind of playpen environment but in bad look (see flipped.png). It is observed that the construction process generate a warning "Extract   bad average roots: 21". 

Could you guys suggest how to fix this issue? Thanks in advance

Combined pointcloud
![combined_plys](https://user-images.githubusercontent.com/49134588/161655145-3f06cb56-cc9c-4092-a89b-3649af5087cf.png)

Mesh Front
![front](https://user-images.githubusercontent.com/49134588/161655095-b38aaa01-beb2-49a9-b6d8-d6a835ce2ba1.png)

Mesh Flipped

![flipped](https://user-images.githubusercontent.com/49134588/161655178-6bc83a84-2c5c-459b-8ad6-50aa6a0bcb77.png)

**Comment by DevAB-Git** [comment]
I did some R&D and observed that normals play a critical role in mesh reconstruction. I created the normals using cloudcompare and then set their orientation using orient_normals_to_align_with_direction. The registered and combined cloud now have normals, apparently aligned (see pic normals_front and normals_back). Consequently, there is some improvement in the flipped mesh but still the front side is a black rectangle. Any help/hint is much appreciated. 

normals_front

![normals_front](https://user-images.githubusercontent.com/49134588/162554495-5d4cbd1c-a2d9-476e-bb4b-2a823f40c8ce.png)

normals_back

![normals_back](https://user-images.githubusercontent.com/49134588/162554503-97bc4e76-ecaf-45ce-91ea-3829cb3a5007.png)


Mesh Flipped
![flipped_with_normals](https://user-images.githubusercontent.com/49134588/162554506-39ddea5c-beb0-408d-a314-5465b999f012.png)

**Comment by Mollylulu** [comment]
@DevAB-Git hi~ I met the same issue as yours. Were you able to find a solution? If you did, I would greatly appreciate it if you could share your solution. Thank you!

**Comment by EwingKang** [comment]
Did you call `SetVertexColors()`? If not, you might want to do so. I can display `open3d::t::geometry::TriangleMesh` color properly with that API. It wont be generated automatically as far as I'm aware.
API: 
https://www.open3d.org/docs/latest/cpp_api/classopen3d_1_1t_1_1geometry_1_1_triangle_mesh.html
